src/ai/backend/manager/models/group.py

```diff
@@ -193,6 +196,48 @@ async def resolve_groups(
     return return_val
 
 
+async def query_suadmin_accessible_groups(
+    db_conn: SAConnection,
+) -> None:
+    """
+    Query all groups.
+    This function returns `None` since super admin can access all groups
+    and fetching all groups from DB is a big overhead.
+    """
+    return None
+
+
+async def query_admin_accessible_groups(
+    db_conn: SAConnection,
+    domain_name: str,
+) -> Iterable[uuid.UUID]:
+    """
+    Query all groups of the domain the admin belongs to.
+    """
+    query = sa.select([groups.c.id]).select_from(groups).where(groups.c.domain_name == domain_name)
+    result = await db_conn.execute(query)
+    return [grp.id for grp in result.fetchall()]
+
+
+async def query_user_accessible_groups(
+    db_conn: SAConnection,
+    user_uuid: uuid.UUID,
+) -> Iterable[uuid.UUID]:
+    """
+    Query all groups the user belongs to.
+    """
+    from .user import users
```

Comment:
Importing package every time when executing function will create an overhead. Let's move this line to outside of this function.